File: 0921_minimum_add_to_make_parentheses_valid/solution.cpp
// LeetCode 0921 - Minimum Add to Make Parentheses Valid
// https://leetcode.com/problems/minimum-add-to-make-parentheses-valid/

#include <string>

class Solution {
public:
    int minAddToMakeValid(std::string s) {
        int openNeed = 0, closeNeed = 0;
        for (char ch : s) {
            if (ch == '(') closeNeed++;
            else if (closeNeed) closeNeed--;
            else openNeed++;
        }
        return openNeed + closeNeed;
    }
};

File: 0921_minimum_add_to_make_parentheses_valid/Solution.swift
// LeetCode 0921 - Minimum Add to Make Parentheses Valid
// https://leetcode.com/problems/minimum-add-to-make-parentheses-valid/

class Solution {
    func minAddToMakeValid(_ s: String) -> Int {
        var openNeed = 0, closeNeed = 0
        for ch in s {
            if ch == "(" { closeNeed += 1 }
            else if closeNeed > 0 { closeNeed -= 1 }
            else { openNeed += 1 }
        }
        return openNeed + closeNeed
    }
}
